src/psyclone/tests/psyir/transformations/intrinsics/maxval2code_trans_test.py

```diff
@@ -0,0 +1,175 @@
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: R. W. Ford, STFC Daresbury Laboratory
+
+'''Module containing tests for the maxval2code transformation.'''
+
+import pytest
+
+from psyclone.psyir.nodes import Reference, ArrayReference
+from psyclone.psyir.symbols import (
+    REAL_TYPE, DataSymbol, INTEGER_TYPE, ArrayType)
+from psyclone.psyir.transformations import (
+    Maxval2CodeTrans, TransformationError)
+
+
+def test_initialise():
+    '''Test that we can create an instance of the transformation and that
+    _INTRINSIC_NAME is set up as expected.
+
+    '''
+    trans = Maxval2CodeTrans()
+    assert isinstance(trans, Maxval2CodeTrans)
+    assert trans._INTRINSIC_NAME == "MAXVAL"
+
+
+def test_loop_body():
+    '''Test that the _loop_body method works as expected, without an array
+    reduction.
+
+    '''
+    trans = Maxval2CodeTrans()
+    i_iterator = DataSymbol("i", INTEGER_TYPE)
+    j_iterator = DataSymbol("j", INTEGER_TYPE)
+    array_iterators = [j_iterator, i_iterator]
+    var_symbol = DataSymbol("var", REAL_TYPE)
+    array_symbol = DataSymbol("array", ArrayType(REAL_TYPE, [10, 10]))
+    array_ref = ArrayReference.create(
+        array_symbol, [Reference(i_iterator), Reference(j_iterator)])
+    result = trans._loop_body(False, array_iterators, var_symbol, array_ref)
+    assert result.debug_string() == (
+        "if (var < array(i,j)) then\n"
+        "  var = array(i,j)\n"
+        "end if\n")
+
+
+def test_loop_body_reduction():
+    '''Test that the _loop_body method works as expected, with an array
+    reduction.
+
+    '''
+    trans = Maxval2CodeTrans()
+    i_iterator = DataSymbol("i", INTEGER_TYPE)
+    j_iterator = DataSymbol("j", INTEGER_TYPE)
+    k_iterator = DataSymbol("k", INTEGER_TYPE)
+    array_iterators = [i_iterator, k_iterator]
+    var_symbol = DataSymbol("var", ArrayType(REAL_TYPE, [10, 10]))
+    array_symbol = DataSymbol("array", ArrayType(REAL_TYPE, [10, 10, 10]))
+    array_ref = ArrayReference.create(
+        array_symbol, [Reference(i_iterator), Reference(j_iterator),
+                       Reference(k_iterator)])
+    result = trans._loop_body(True, array_iterators, var_symbol, array_ref)
+    assert result.debug_string() == (
+        "if (var(i,k) < array(i,j,k)) then\n"
+        "  var(i,k) = array(i,j,k)\n"
+        "end if\n")
+
+
+def test_init_var():
+    '''Test that the _init_var method works as expected.'''
+    trans = Maxval2CodeTrans()
+    var_symbol = DataSymbol("var", REAL_TYPE)
+    result = trans._init_var(var_symbol)
+    # As 'tiny' is not yet part of an expression, the 'debug_string()'
+    # method incorrectly assumes it is a call.
+    assert result.debug_string() == "call TINY(var)\n"
+
+
+def test_str():
```

Comment:
Are these two tests needed here - could they be moved to the mms test file?

Reply:
I would argue what I argued before. They are checking that the particular transformation information is added to the str() output which is supplied by this class.

Reply:
And I'd argue what I argued before (but I've decided to let it go...[no, really])